core/src/test/java/ai/timefold/solver/core/impl/solver/BestSolutionHolderTest.java

```diff
@@ -186,6 +189,34 @@ void problemChangeBarrageIntermediateBestSolutionConsumer() throws InterruptedEx
 
     }
 
+    @Test
+    void testCloningBetweenPhases() throws ExecutionException, InterruptedException {
+        var solverConfig = new SolverConfig()
+                .withSolutionClass(TestdataSolution.class)
+                .withEntityClasses(TestdataEntity.class)
+                .withEasyScoreCalculatorClass(TestdataEasyScoreCalculator.class);
+
+        try (var solverManager = SolverManager.<TestdataSolution, UUID> create(solverConfig, new SolverManagerConfig())) {
+            var problem = TestdataSolution.generateSolution(2, 2);
+            problem.getEntityList().forEach(entity -> entity.setValue(null));
+            var countDownLatch = new CountDownLatch(1);
+            var solution = solverManager.solveBuilder()
+                    .withProblemId(UUID.randomUUID())
+                    .withProblem(problem)
+                    .withBestSolutionConsumer(
+                            intermediate -> {
+                                intermediate.getEntityList().add(new TestdataEntity("Bad Entity 3"));
+                                countDownLatch.countDown();
+                            })
+                    .withConfigOverride(new SolverConfigOverride<TestdataSolution>()
+                            .withTerminationConfig(new TerminationConfig().withMoveCountLimit(100L)))
+                    .run()
+                    .getFinalBestSolution();
+            countDownLatch.await();
+            assertThat(solution.getEntityList()).hasSize(2);
```

Comment:
This deserves a comment explaining why the solution only has 2 elements.
(Essentially a comment to explain what the test tries to do.)